beacon/goclient/goclient.go

```diff
@@ -215,9 +250,44 @@ func New(
 	// Start automatic expired item deletion for attestationDataCache.
 	go client.attestationDataCache.Start()
 
+	if err := client.launchEventListener(opt.Context); err != nil {
+		return nil, errors.Wrap(err, "failed to launch head event listener")
+	}
+
 	return client, nil
 }
 
+func (gc *GoClient) launchEventListener(ctx context.Context) error {
```

Comment:
launchBlockRootListener should be a less confusing name, bc this name sounds like its generic for events and  should be in `events.go` (but its not)